packages/client/screens/WalletSettingsScreen.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import { SettingsStackScreenProps } from "../types";
+import { Button } from "native-base";
+import { Alert, StyleSheet } from "react-native";
+import { View } from "../components/Themed";
+import { AuthContext } from "../components/contexts/AuthContext";
+import { WalletsGateway } from "../gateways/wallets_gateway";
+import { getCurrencyType } from "@cryptify/common/src/utils/currency_utils";
+
+type Props = SettingsStackScreenProps<"WalletSettingsScreen">;
+
+export default function WalletSettingsScreen({ navigation, route }: Props) {
+    const { address, currencyType } = route.params;
+    const { token, user } = React.useContext(AuthContext);
+    const walletsGateway = new WalletsGateway();
+
+    function handleDeleteWallet(): void {
+        Alert.alert("Do you want to remove this wallet?", "You cannot undo this action.", [
+            { text: "Cancel", style: "cancel" },
+            {
+                text: "Remove",
+                style: "destructive",
+                onPress: () => {
+                    walletsGateway.deleteWallet(
+                        { userId: user.id, address: address, currencyType: getCurrencyType(address) },
```

Comment:
CurrencyType is passed as a prop here

```suggestion
                        { userId: user.id, address: address, currencyType },
```

Reply:
currencyType was retiring the value "ETH" which was not recognized as a currency type in the api. This solves the issue.

Reply:
Yep because you guys had mixed up the currency type and the iso code, I already fixed it you just gotta rebase then currencyType prop will give the correct value